src/jit/importer.cpp

```diff
@@ -18376,110 +18387,33 @@ void Compiler::impDevirtualizeCall(GenTreeCall*            call,
     }
 
 #if defined(DEBUG)
-
     // Bail if devirt is disabled.
     if (JitConfig.JitEnableDevirtualization() == 0)
     {
         return;
     }
 
+    const bool doPrint = JitConfig.JitPrintDevirtualizedMethods() == 1;
```

Comment:
`doPrint` only exists in `DEBUG`. There is no release mode diagnostic stream for devirtualization.

Not sure if devirtualization warrants an VM/ETW callback/event. Inlining needs it because other runtime subsystems are sensitive to inlining -- for instance rejitting needs to know what methods have been inlined.

If we do add a notification mechanism it will traffic only in handles and string literals to avoid any need for dynamic string allocation.